openedx/core/djangoapps/theming/storage.py

```diff
@@ -17,30 +23,22 @@ class ComprehensiveThemingAwareMixin(object):
     """
     def __init__(self, *args, **kwargs):
         super(ComprehensiveThemingAwareMixin, self).__init__(*args, **kwargs)
-        COMP_THEME_DIR = getattr(settings, "COMP_THEME_DIR", "")  # pylint: disable=invalid-name
-        if not COMP_THEME_DIR:
+        path_theme = Path(settings.COMPREHENSIVE_THEMING_DIRECTORY)
+        if not path_theme:
             self.theme_location = None
             return
-
-        if not isinstance(COMP_THEME_DIR, basestring):
-            raise ImproperlyConfigured("Your COMP_THEME_DIR setting must be a string")
-
-        PROJECT_ROOT = getattr(settings, "PROJECT_ROOT", "")  # pylint: disable=invalid-name
-        if PROJECT_ROOT.endswith("cms"):
-            component = "studio"
-        else:
-            component = "lms"
-        self.theme_location = os.path.join(COMP_THEME_DIR, component, "static")
+        paths = get_paths(path_theme)
+        self.theme_location = paths['static']
 
     @property
     def prefix(self):
         """
         This is used by the ComprehensiveThemeFinder in the collection step.
         """
-        COMP_THEME_DIR = getattr(settings, "COMP_THEME_DIR", "")  # pylint: disable=invalid-name
```

Comment:
Why are we disabling pylint violations instead of just fixing them?

I can tell that you're mimicking the naming convention from `settings`,
but these aren't constants. They're local variables.

It's fine to memoize them, but we should name them properly.